user_tools/src/spark_rapids_tools/tools/additional_heuristics.py

```diff
@@ -68,18 +70,43 @@ def _apply_heuristics(self, app_ids: list) -> pd.DataFrame:
         else:
             for app_id in app_ids:
                 app_id_path = os.path.join(metrics_path, app_id)
+                # Apply heuristics and determine if the application should be skipped.
+                # Note: `should_skip` flag can be a combination of multiple heuristic checks.
+                should_skip = False
+                # 1. Apply heristics based on total core seconds
                 try:
-                    # Apply heuristics and determine if the application should be skipped.
-                    # Note: `should_skip` flag can be a combination of multiple heuristic checks.
-                    should_skip, reason = self.heuristics_based_on_spills(app_id_path)
+                    should_skip, reason = self.heuristics_based_on_total_core_seconds(app_id)
                 except Exception as e:  # pylint: disable=broad-except
-                    should_skip = False
-                    reason = f'Cannot apply heuristics for qualification. Reason - {type(e).__name__}:{e}'
+                    reason = ' Cannot apply total core seconds heuristics for qualification. Reason - ' + \
+                        f'{type(e).__name__}:{e}.'
+                    self.logger.error(reason)
+                # 2. Apply heristics based on spilling
+                try:
+                    should_skip_spill, reason_spill = self.heuristics_based_on_spills(app_id_path)
+                    should_skip = should_skip or should_skip_spill
+                    reason = reason + reason_spill
+                except Exception as e:  # pylint: disable=broad-except
+                    reason = reason + ' Cannot apply spill heuristics for qualification. Reason - ' + \
+                        f'{type(e).__name__}:{e}.'
                     self.logger.error(reason)
                 result_arr.append([app_id, should_skip, reason])
 
         return pd.DataFrame(result_arr, columns=self.props.get_value('resultCols'))
 
+    def heuristics_based_on_total_core_seconds(self, app_id: str) -> (bool, str):
+        """
+        Apply heuristics based on total core seconds to determine if the app can be accelerated on GPU.
+        """
+        # Load app total core seconds from qualidication summary dataframe
+        app_qual_output = self.qual_summary[self.qual_summary['App ID'] == app_id]
+        total_core_seconds = app_qual_output['Total Core Seconds'].astype(int).iloc[0]
+        # Threshold is total core seconds of an n1-standard-8 instance running for one day
```

Comment:
Can we include the comment in the config file where we define the threshold value?

Reply:
fixed, thanks!